src/main/js/field/TableDisplay.js

```diff
@@ -295,68 +323,87 @@ class TableDisplay extends Component {
+        const { onConfigCopy } = this.props;
         const callback = () => this.setState({
             currentRowSelected,
             showConfiguration: true
         });
-        this.props.onConfigCopy(currentRowSelected, callback);
+        onConfigCopy(currentRowSelected, callback);
     }
 
     copyButtonClick(cell, row) {
-        return (<IconTableCellFormatter
-            handleButtonClicked={this.copyButtonClicked}
-            currentRowSelected={row}
-            buttonIconName="copy"
-            buttonText="Copy"
-        />);
+        return (
+            <IconTableCellFormatter
+                handleButtonClicked={this.copyButtonClicked}
+                currentRowSelected={row}
+                buttonIconName="copy"
+                buttonText="Copy"
+            />
+        );
     }
 
     render() {
         const tableColumns = this.createTableColumns();
-        if (this.props.enableEdit) {
-            tableColumns.push(<TableHeaderColumn
-                dataField=""
-                width="48"
-                columnClassName="tableCell"
-                dataFormat={this.editButtonClick}
-                thStyle={{ textAlign: 'center' }}
-            >Edit</TableHeaderColumn>);
-        }
-        if (this.props.enableCopy) {
-            tableColumns.push(<TableHeaderColumn
-                dataField=""
-                width="48"
-                columnClassName="tableCell"
-                dataFormat={this.copyButtonClick}
-                thStyle={{ textAlign: 'center' }}
-            >Copy</TableHeaderColumn>);
-        }
-
+        const { showDelete } = this.state;
         const {
-            selectRowBox, sortName, sortOrder, autoRefresh, tableMessage, newButton, deleteButton, data, tableSearchable, inProgress
+            selectRowBox, sortName, sortOrder, autoRefresh, tableMessage, newButton, deleteButton, data,
+            tableSearchable, enableEdit, enableCopy, inProgress, tableRefresh, refreshData
         } = this.props;
+        if (enableEdit) {
+            const editColumn = (
+                <TableHeaderColumn
+                    dataField=""
+                    width="48"
+                    columnClassName="tableCell"
+                    dataFormat={this.editButtonClick}
+                    thStyle={{ textAlign: 'center' }}
+                >
+                    Edit
+                </TableHeaderColumn>
+            );
+            tableColumns.push(editColumn);
+        }
+        if (enableCopy) {
+            const copyColumn = (
+                <TableHeaderColumn
+                    dataField=""
+                    width="48"
+                    columnClassName="tableCell"
+                    dataFormat={this.copyButtonClick}
+                    thStyle={{ textAlign: 'center' }}
+                >
+                    Copy
+                </TableHeaderColumn>
+            );
+            tableColumns.push(copyColumn);
+        }
```

Comment:
Instead of creating 2 complete differnt TableHeaderColumn objects, we should change this to create a single one, but pass it different values to dataFormat and children. This would cut down on needing to change multiple places if we need to change the look of this column.

Reply:
we push it into an array of table columns.  They need to be separate object instances.  I will create a function that will return a new instance.